README.md

```diff
@@ -58,6 +58,38 @@ If this is your first time running the helper script, make sure to run the `Init
 
 If the pre-commit hook is not letting you commit something and you want to override it. You can ignore the sqlfluff lint hook by running your `git commit` with the `--no-verify` flag.
 
+## Containerizing dbt-cta
+There is a _Dockerfile_ at the root of this repo that can be used to create a dbt-cta image. This image can be used to run dbt as a container, which makes it possible to run dbt in either Docker or Kubernetes. There is a Makefile that can be used to simplify the process of building and running the dbt-cta image locally for testing!
+
+### Makefile Requirements
+- A Google Cloud Platform(GCP) Project with Cloud Build and [Artifact Registry](https://cloud.google.com/artifact-registry/docs/docker) enabled. The Makefile works by using GCP Cloud Build to build and deploy the image to Google's Artifact Registry. 
+- Setup Docker Authentication to Artifact Registry
+    - [Documentation from GCP](https://cloud.google.com/artifact-registry/docs/docker/authentication)
+    - I've had no issues using the [gcloud cli credential helper](https://cloud.google.com/artifact-registry/docs/docker/authentication#gcloud-helper)
+- If you are running the dbt locally, you mush have Docker up and running on your computer.
+
+### Using the Makefile
+Below are the commands you can run with the Makefile. To view a summary of these on your CLI, run `make help`
+
+- `make help` - prints out a help menu with all the available commands
+- `make build PROJECT_ID=myprojectid` - Starts a Cloud Build run to build and deploy an image. It uses the current Git HEAD hash as the tag.
+- `make tag PROJECT_ID=myprojectid` - Adds the _latest_ tag to Artifact tagged with the current Git Hash.
+    - Optionally, you can specify a different tag to add and a specific Git Hash to apply the tag to.
+    Example: `make tag PROJECT_ID=myprojectid TAG=0.0.1 GIT_HASH=somecommithash`
+- `make run PROJECT_ID=myprojectid` - pulls the image with the current git hash tag and runs dbt locally on Docker. 
+    - NOTE: You must have a `.env` file at the root of this repo for this to work. The `.env` file should have the following Variables set.
+    ```shell
+    CTA_PROJECT_ID=source_project_id
+    CTA_DATASET_ID=source_dataset_id
+    PARTNER_PROJECT_ID=destination_project_id
+    PARTNER_DATASET_ID=destination_dataset_id
+    SYNC_NAME=sync_name  # Example: actblue
+    DBT_TARGET="cta"  # Can be any available target in profiles.yml
+    DBT_SELECT="tag:cta"  # Specify what models should be selected, can specify tags, model names, folders, etc..
+    ```
```

Comment:
One thing I've done before is to create an `.env.example` file that's kept up to date with the structure. Values or fake, or strings that provide a clue as to where to get potentially sensitive values from.